pathways/simulation.py

```diff
@@ -183,40 +196,81 @@ def simulation(
         avg_intercepted_infestation_rate = 0
         pct_pest_unreported_if_detection = 0
 
-    return types.SimpleNamespace(
-        missing=missing,
-        false_neg=false_neg,
-        intercepted=success_rates.true_positive,
-        num_inspections=num_inspections,
-        total_num_boxes=total_num_boxes,
-        total_num_stems=total_num_stems,
-        avg_boxes_opened_completion=total_boxes_opened_completion / num_shipments,
-        avg_boxes_opened_detection=total_boxes_opened_detection / num_shipments,
-        pct_boxes_opened_completion=(
-            (total_boxes_opened_completion / total_num_boxes) * 100
-        ),
-        pct_boxes_opened_detection=(
-            (total_boxes_opened_detection / total_num_boxes) * 100
-        ),
-        avg_stems_inspected_completion=total_stems_inspected_completion / num_shipments,
-        avg_stems_inspected_detection=total_stems_inspected_detection / num_shipments,
-        pct_stems_inspected_completion=(
-            (total_stems_inspected_completion / total_num_stems) * 100
-        ),
-        pct_stems_inspected_detection=(
-            (total_stems_inspected_detection / total_num_stems) * 100
-        ),
-        pct_pest_unreported_if_detection=pct_pest_unreported_if_detection,
-        true_infestation_rate=true_infestation_rate / num_shipments,
-        max_missed_infestation_rate=max_missed_infestation_rate,
-        avg_missed_infestation_rate=avg_missed_infestation_rate,
-        max_intercepted_infestation_rate=max_intercepted_infestation_rate,
-        avg_intercepted_infestation_rate=avg_intercepted_infestation_rate,
-        false_negative_present=false_negative_present,
-        true_positive_present=true_positive_present,
-        total_intercepted_pests=total_intercepted_pests,
-        total_missed_pests=total_missed_pests,
-    )
```

Comment:
You can just assign this to a variable and assign the details later if needed.

```
ret = types.SimpleNamespace(a=1)
ret.details = "abc"
```

Also according to documentation of SimpleNamespace _you can add and remove attributes_.

Reply:
Ah ok! That's much better, thanks.